tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debug_tensor_value.ts

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {DTYPE_ENUM_TO_NAME} from '../tf_dtypes';
+import {DebugTensorValue, TensorDebugMode} from './debugger_types';
+
+/**
+ * Parse a number array that represents debugging summary of an instrumented
+ * tensor value.
+ *
+ * @param tensorDebugMode Tensor-debug mode.
+ * @param array The array of number that represents various aspect of the
+ *   instrumented tensor. The semantics of the numbers are determined by
+ *   `tensorDebugMode`.
+ * @returns A DebugTensorValue object with the same information as
+ *   carried by `array`, but represented in a more explicit fashion.
+ *   For numbers that represent breakdown of numeric values by type
+ *   (e.g., counts of -inf, +inf and nan), the corresponding fields
+ *   in the returned object will be defined only if the count is non-zero.
+ */
+export function parseDebugTensorValue(
```

Comment:
I am not too convinced about the performance argument. Protobuf, sure, may not be as efficient at representing such values as what you did here but it surely is closer to this than, say, JSON (jspb encodes values as array of arrays; it does not transfer binaries). 

I am still okay with this because jspb dependency (closure version) on FE is quite not pleasant.